app/validators/keywords_form_validator.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+class KeywordsFormValidator < ActiveModel::Validator
+  def validate(form)
+    @form = form
+    validate_file
+  end
+
+  private
+
+  attr_reader :form
+
+  def validate_file
+    if !keywords_file
+      add_error(I18n.t('keywords.upload.invalid_file'))
+    elsif extension_valid?
+      add_error(I18n.t('keywords.upload.invalid_size')) unless size_valid?
+    else
+      add_error(I18n.t('keywords.upload.invalid_extension'))
```

Comment:
Can we refactor this a bit? I think it is fine not to have `else` block.
```suggestion
    if !keywords_file
      add_error(I18n.t('keywords.upload.invalid_file'))
    elsif !size_valid?
      add_error(I18n.t('keywords.upload.invalid_size')) 
    elsif !extension_valid?
      add_error(I18n.t('keywords.upload.invalid_extension'))
    end
```

Reply:
@hanam1ni with the new logic, we are checking the `size_valid?` before `extension_valid?`, but if the user upload a `non csv file`, then there will be an error because the function `size_valid?` is `CSV.read(keywords_file.path)`

I will try to refactor the current logic :D

Reply:
How about we check for an extension before checking the size?
```ruby
elsif !extension_valid?
      add_error(I18n.t('keywords.upload.invalid_extension'))
elsif !size_valid?
      add_error(I18n.t('keywords.upload.invalid_size')) 
end
```

Reply:
Fixed in [`ff47954` (#42)](https://github.com/suho/google-search-ruby/pull/42/commits/ff4795420581e0777bbcdd32fe6c14fa6c7bd059)